packages/terra-i18n-plugin/src/I18nAggregatorPlugin.js

```diff
@@ -23,9 +23,16 @@ function getDirectories(srcPath, inputFileSystem) {
   return inputFileSystem.readdirSync(srcPath).filter(file => inputFileSystem.statSync(path.join(srcPath, file)).isDirectory());
 }
 
-function aggregateDirectory(languageMessages, currentDirectory, inputFileSystem) {
-  // Check the directory for translations
-  const translationsDirectory = path.resolve(currentDirectory, 'translations');
+// Find all descendant directories by name under a root directory and return an array
+function findDirectories(dirName, rootDir, inputFileSystem) {
+  const result = [];
+  const subDirs = getDirectories(rootDir, inputFileSystem);
+  result.push(subDirs.filter(dir => dir === dirName).map(dir => path.join(rootDir, dir)));
+  subDirs.forEach(dir => result.push(findDirectories(dirName, path.join(rootDir, dir), inputFileSystem)));
```

Comment:
Using findDirectories within the findDirectories function seems odd to me. I think adding some more doc around what all this is doing would make it more clear and easier to maintain in the future.

Reply:
Updated with more docs